File: src/main/java/nuclearcoder/discordbot/command/CommandManager.java
package nuclearcoder.discordbot.command;

import nuclearcoder.util.Config;
import sx.blah.discord.api.events.IListener;
import sx.blah.discord.handle.impl.events.MessageReceivedEvent;

public interface CommandManager extends IListener<MessageReceivedEvent> {

    String COMMAND_PREFIX = Config.get("command_prefix");

    void initCommands();

    void clearCommands();

    void handle(MessageReceivedEvent event);

    Command getCommand(String label);

    void putCommand(String label, Command command);

    void removeCommand(String label);

}

File: src/main/java/nukebot/command/CommandManager.kt
package nukebot.command

import nukeutils.Config
import sx.blah.discord.api.events.IListener
import sx.blah.discord.handle.impl.events.MessageReceivedEvent

interface CommandManager : IListener<MessageReceivedEvent> {

    companion object {
        val COMMAND_PREFIX: String = Config["command_prefix"]
    }

    fun initCommands()

    fun clearCommands()

    override fun handle(event: MessageReceivedEvent)

    fun getCommand(label: String): Command?

    fun putCommand(label: String, command: Command)

    fun removeCommand(label: String)

}